src/python/pants/backend/python/macros/python_requirements_test.py

```diff
@@ -104,6 +105,55 @@ def test_requirements_txt(rule_runner: RuleRunner) -> None:
     )
 
 
+def test_multiple_versions(rule_runner: RuleRunner) -> None:
+    """This tests that we correctly create a new python_requirement_library for each unique
+    dependency name in a requirements.txt file, grouping duplicated dependency names to handle
+    multiple requirement strings per PEP 508."""
+
+    assert_python_requirements(
+        rule_runner,
+        "python_requirements(module_mapping={'ansicolors': ['colors']})",
+        dedent(
+            """\
+            Django>=3.2
+            Django==3.2.7
+            confusedmonkey==86
+            repletewateringcan>=7
+            """
+        ),
+        expected_file_dep=PythonRequirementsFile(
+            {"sources": ["requirements.txt"]},
+            Address("", target_name="requirements.txt"),
+        ),
+        expected_targets=[
+            PythonRequirementLibrary(
+                {
+                    "dependencies": [":requirements.txt"],
+                    "requirements": [
+                        Requirement.parse("Django>=3.2"),
```

Comment:
I'm not sure what the Pants rendering is, but Pex will say this for this example:
```$ pex Django>=3.2 Django==3.2.7
ERROR: Double requirement given: Django==3.2.7 (already in Django, name='Django')
...
```
That error message works in this case since I typed the dup reqs in directly. In a Pants resolve context though this  might not be awesome. Although we don't have the technology to calculate disjointness on board, we could at least fail any group where all requirements don't have markers which we _know_ will ead to dup ahead of time. Maybe worth a follow-up to complicate our code to provide a better error message some of the time.

Reply:
A simple improvement is to warn or error if there is >1 requirement for same project_name and they do not all have `Requirement.extras`, meaning no environment markers.

Reply:
Extras and environment markers are (or should be) disjoint in this context. What is this supposed to mean?:
```
"requests[security]>=2" "requests<2" 
```
That should fail just like this should fail:
```
"requests>=2" "requests<2" 
```
Its only if both requests have environment markers and those markers pick out disjoint sets that the resolve makes sense.

Reply:
@jsirois Just to confirm -- is there any follow-up on this PR?

Reply:
No changes to this PR. I was just suggesting potential follow-up in a subsequent PR.

Reply:
Oh you're right. I mixed up my terms. Sorry! I meant require environment markers, not extras.